src/history.js

```diff
@@ -343,9 +343,11 @@ class History extends Emitter {
       }
     }
 
-    root.prune()
+    if (root !== this.root) {
+      this.root = root
+      root.prune()
```

Comment:
Just a small thing. root.prune() would get called every time, even if the root didn't change.